### PR TITLE
PS-9003 Add pro-builds and FIPS documentation (8.0)

### PR DESCRIPTION
**Pull request by alina-derkach-oaza**
modified:   docs/compile-percona-server.md
	new file:   docs/fips.md
	new file:   docs/install-pro.md
	modified:   docs/installation.md
	new file:   docs/psmysql-pro.md
	modified:   docs/quickstart-overview.md
	modified:   docs/source-tarball.md
	new file:   docs/upgrade-pro.md
	modified:   mkdocs-base.yml

### CODE REVIEW

File: docs/upgrade-pro.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Upgrade to Percona Server for MySQL Pro
+
+Are you a Percona Customer already and are you ready to enjoy all the [benefits of Percona Server for MySQL Pro](../psmysql-pro.md)? 
+
+This document provides instructions on how you can upgrade from Percona Server for MySQL Basic to Percona Server for MySQL Pro.
+
+## Preconditions 
+
+Request the access to the pro repository from Percona Support. You will receive the client ID and the access token.
```
Comment (patrickbirch) [inline docs/upgrade-pro.md]: client ID and access token which you use when downloading the packages.

File: docs/upgrade-pro.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Upgrade to Percona Server for MySQL Pro
+
+Are you a Percona Customer already and are you ready to enjoy all the [benefits of Percona Server for MySQL Pro](../psmysql-pro.md)? 
+
+This document provides instructions on how you can upgrade from Percona Server for MySQL Basic to Percona Server for MySQL Pro.
```
Comment (patrickbirch) [inline docs/upgrade-pro.md]: Are we calling it Percona Server for MySQL Basic or just Percona Server for MySQL
I'd ask Val to clarify.

File: docs/psmysql-pro.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Percona Server for MySQL Pro
+
+Percona Server for MySQL Pro includes [solutions](#solutions) typically demanded by large enterprises. Percona Server for MySQL Pro is wrapped in packages created and tested by Percona. These packages are supported only for Percona Customers with a subscription.
```
Comment (patrickbirch) [inline docs/psmysql-pro.md]: MySQL Pro is wrapped in packages?
Maybe
MySQL Pro contains packages created and tested by Percona.

File: docs/install-pro.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Install Pro packages of Percona Server for MySQL
+
+This document provides guidelines how to install Pro packages of Percona Server for MySQL from Percona repositories. [Learn more about Percona Server for MySQL Pro :material-arrow-top-right:](../psmysql-pro.md).
+
+## Procedure
+
+1. Request the access to the pro repository from Percona Support. You will receive the client ID and the access token.
```
Comment (patrickbirch) [inline docs/install-pro.md]: client ID and access token which you use when downloading the packages.

File: docs/fips.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# FIPS compliance
+
+FIPS (Federal Information Processing Standard) is the US government computer security standard for cryptography modules that include both hardware and software components. Percona Server for MySQL supports FIPS certified module for OpenSSL, enabling US organizations to introduce FIPS-compliant encryption and thus meet the requirements towards data security.
```
Comment (patrickbirch) [inline docs/fips.md]: The Federal Information Processing Standards (FIPS) are a set of U.S. government standards that ensure the security of computer systems for non-military government agencies and contractors. These standards specify how to perform cryptographic operations, such as encryption, hashing, and digital signatures. FIPS mode is a mode of operation that enforces these standards and rejects any non-compliant algorithms or parameters.

Comment (patrickbirch) [inline docs/fips.md]: Percona Server for MySQL can run in FIPS mode if a FIPS-enabled OpenSSL library and FIPS Object Module are available at runtime or if compiled using a compliant version of OpenSSL.

File: docs/fips.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# FIPS compliance
+
+FIPS (Federal Information Processing Standard) is the US government computer security standard for cryptography modules that include both hardware and software components. Percona Server for MySQL supports FIPS certified module for OpenSSL, enabling US organizations to introduce FIPS-compliant encryption and thus meet the requirements towards data security.  
+
+Percona Server for MySQL implements the same level of FIPS support as MySQL. You can also receive this functionality by [building Percona Server for MySQL from source code](compile-percona-server.md).
+
+See [Configure MySQL for FIPS](https://dev.mysql.com/doc/refman/8.0/en/fips-mode.html) in MySQL documentation for configuration guidelines.
```
Comment (patrickbirch) [inline docs/fips.md]: Ask what the variables are and the parameters.

Comment (alina-derkach-oaza) [reply docs/fips.md]: Updated

File: docs/fips.md (hunk 5 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: Updated

Comment (alina-derkach-oaza) [reply docs/fips.md]: Updated

File: docs/install-pro.md (hunk 4 above)
Comment (alina-derkach-oaza) [reply docs/install-pro.md]: Updated

File: docs/psmysql-pro.md (hunk 3 above)
Comment (alina-derkach-oaza) [reply docs/psmysql-pro.md]: updated

File: docs/upgrade-pro.md (hunk 1 above)
Comment (alina-derkach-oaza) [reply docs/upgrade-pro.md]: Updated

File: docs/fips.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    * `1` - enables FIPS mode
+    * `2` - enables `strict` FIPS mode
+
+* The `--ssl-fips-mode` client option controls whether a given MySQL client operates in FIPS mode. This setting does not change the server setting. This option is disabled by default.
+
+    The `--ssl-fips-mode` client option has these values:
+
+    * `OFF` - disables FIPS mode
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
```
Comment (patrickbirch) [inline docs/fips.md]: Include the STRICT definition in the server version.

File: docs/fips.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+     
+    Note that the MySQL server does not read the OpenSSL configuration file unless the `--ssl-fips-mode` client option is set to `ON` or `STRICT`. The `--ssl-fips-mode` client option is read-only and cannot be changed at runtime.
+  
+    To enable the FIPS mode, pass `--ssl-fips-mode=ON` to mysqld as a command line argument. Ignore the warning that the `--ssl-fips-mode` client option is deprecated. To ensure that the FIPS mode is enabled, check that the error log contains the following message:
```
Comment (patrickbirch) [inline docs/fips.md]: Make this a separate section

## Check FIPS Mode

File: docs/fips.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# FIPS compliance
+
+The Federal Information Processing Standards (FIPS) are a set of U.S. government standards that ensure the security of computer systems for non-military government agencies and contractors. These standards specify how to perform cryptographic operations, such as encryption, hashing, and digital signatures. FIPS mode is a mode of operation that enforces these standards and rejects any non-compliant algorithms or parameters.
+
+Percona Server for MySQL implements the same level of FIPS support as MySQL. Percona Server for MySQL can run in FIPS mode if a FIPS-enabled OpenSSL library and FIPS Object Module are available at runtime or if compiled using a compliant version of OpenSSL. You can also receive this functionality by [building Percona Server for MySQL from source code](compile-percona-server.md).
```
Comment (patrickbirch) [inline docs/fips.md]: compiled using a FIPS-validated version of OpenSSL.

Comment (alina-derkach-oaza) [reply docs/fips.md]: fixed

File: docs/fips.md (hunk 8 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: updated

File: docs/fips.md (hunk 7 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: added

File: docs/fips.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    
+    ### Check FIPS mode
+
+    To ensure that the FIPS mode is enabled, check that the error log contains the following message:
+
+    ```{.text .no-copy}
+    A FIPS-approved version of the OpenSSL cryptographic library has been detected in the operating system with a properly configured FIPS module available for loading. Percona Server for MySQL will load this module and run in FIPS mode.
+    ```
+
+Find more information in the [Configure MySQL for FIPS](https://dev.mysql.com/doc/refman/8.0/en/fips-mode.html) document.
```
Comment (patrickbirch) [inline docs/fips.md]: You can remove this link to MySQL.

Comment (alina-derkach-oaza) [reply docs/fips.md]: removed

File: docs/fips.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
+# FIPS compliance
+
+The Federal Information Processing Standards (FIPS) are a set of U.S. government standards that ensure the security of computer systems for non-military government agencies and contractors. These standards specify how to perform cryptographic operations, such as encryption, hashing, and digital signatures. FIPS mode is a mode of operation that enforces these standards and rejects any non-compliant algorithms or parameters.
+
+Percona Server for MySQL implements the same level of FIPS support as MySQL. Percona Server for MySQL can run in FIPS mode if a FIPS-enabled OpenSSL library and FIPS Object Module are available at runtime or if compiled using a FIPS-validated version of OpenSSL. You can also receive this functionality by [building Percona Server for MySQL from source code](compile-percona-server.md).
+
+Percona Server for MySQL uses the same variables and values as MySQL. Percona Server for MySQL enables control of FIPS mode on the server side and the client side:
+
+* The `ssl_fips_mode` system variable controls whether the server operates in FIPS mode. This variable is disabled by default.
```
Comment (percona-ysorokin) [inline docs/fips.md]: As this system variable became deprecated and read-only in MySQL8.0.34 it no longer "controls" whether the server operates in FIPS mode, just "shows".
To "control", the "--ssl-fops-mode" server command line option should be used

File: docs/fips.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+    The `ssl_fips_mode` system variable has these values:
+
+    * `0` - disables FIPS mode
+    * `1` - enables FIPS mode
+    * `2` - enables `strict` FIPS mode. This value provides more restrictions than the  `1 ` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `1` or `2` depends on the OpenSSL version.
+
+* The `--ssl-fips-mode` client option controls whether a given MySQL client operates in FIPS mode. This setting does not change the server setting. This option is disabled by default.
+
+    The `--ssl-fips-mode` client option has these values:
```
Comment (percona-ysorokin) [inline docs/fips.md]: The same set of values for `--ssl-fips-mode` is used for both server and client command line option.

File: docs/fips.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+    * `OFF` - disables FIPS mode
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
+     
+    Note that the MySQL server does not read the OpenSSL configuration file unless the `--ssl-fips-mode` client option is set to `ON` or `STRICT`. The `--ssl-fips-mode` client option is read-only and cannot be changed at runtime.
```
Comment (percona-ysorokin) [inline docs/fips.md]:
```suggestion
    Note that whether MySQL server operates in FIPS mode does not depend on which crypto module (regular or FIPS-approved) is configured as the default in the OpenSSL configuration file.  The server always respects the value of `--ssl-fips-mode` server command line option (`OFF`, `ON`, or `STRICT`). The `ssl_fips_mode` global system variable is read-only and cannot be changed at runtime.
```

File: docs/upgrade-pro.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        enabled=1
+        gpgkey = https://repo.percona.com/yum/PERCONA-PACKAGING-KEY
+        ```
+
+3. Install Percona Server for MySQL PRO packages
+
+    === "On Debian and Ubuntu"
+
+        ```{.bash .data-prompt="$"}
+        $ sudo apt install -y percona-server-server-pro
```
Comment (percona-ysorokin) [inline docs/upgrade-pro.md:56]: Shouldn't we explicitly remove normal (non-pro) `percona-server-server` packages first here?

File: docs/upgrade-pro.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+        ```
+    === "On RHEL and derivatives"
+
+        ```{.bash .data-prompt="$"}
+        $ sudo yum install -y percona-server-server-pro --allowerasing
```
Comment (percona-ysorokin) [inline docs/upgrade-pro.md]: Likewise

File: docs/fips.md (hunk 11 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: updated

File: docs/fips.md (hunk 13 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: updated

File: docs/fips.md (hunk 12 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: updated

File: docs/upgrade-pro.md (hunk 14 above)
Comment (alina-derkach-oaza) [reply docs/upgrade-pro.md:56]: @eleonora Could you please comment on this?

File: docs/upgrade-pro.md (hunk 15 above)
Comment (alina-derkach-oaza) [reply docs/upgrade-pro.md]: @Eleonora Could you please comment on this?

File: docs/fips.md [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+    The `--ssl-fips-mode` client/server option has these values:
+
+    * `OFF` - disables FIPS mode
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
+     
+    The MySQL server operation in FIPS mode does not depend on which crypto module (regular or FIPS-approved) is set as the default in the OpenSSL configuration file. The server always respects the value of `--ssl-fips-mode` server command line option (`OFF`, `ON`, or `STRICT`). The `ssl_fips_mode` global system variable is read-only and cannot be changed at runtime.
+  
+    To enable the FIPS mode, pass `--ssl-fips-mode=ON` to mysqld as a command line argument. Ignore the warning that the `--ssl-fips-mode` client option is deprecated.
```
Comment (eleo007) [inline docs/fips.md]: Or set in configuration file.

File: docs/fips.md [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
+     
+    The MySQL server operation in FIPS mode does not depend on which crypto module (regular or FIPS-approved) is set as the default in the OpenSSL configuration file. The server always respects the value of `--ssl-fips-mode` server command line option (`OFF`, `ON`, or `STRICT`). The `ssl_fips_mode` global system variable is read-only and cannot be changed at runtime.
+  
+    To enable the FIPS mode, pass `--ssl-fips-mode=ON` to mysqld as a command line argument. Ignore the warning that the `--ssl-fips-mode` client option is deprecated. 
+    
+    ### Check FIPS mode
+
+    To ensure that the FIPS mode is enabled, check that the error log contains the following message:
```
Comment (eleo007) [inline docs/fips.md]: when log-error-verbosity=3

File: docs/install-pro.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+
+## Procedure
+
+1. Request the access to the pro repository from Percona Support. You will receive the client ID and the access token which you use when downloading the packages.
+
+2. Configure the repository
+
+    === "On Debian and Ubuntu"
+
+        1. Create the `/etc/apt/sources.list.d/psmysql-pro.list` configuration file with the following contents
```
Comment (eleo007) [inline docs/install-pro.md]: Before this: download percona gpg key and update `trusted.gpg.d`:
```
wget https://github.com/percona/percona-repositories/raw/main/deb/percona-keyring.gpg 
sudo cp percona-keyring.gpg /etc/apt/trusted.gpg.d/
```

File: docs/install-pro.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+
+            ```ini title="/etc/apt/sources.list.d/psmysql-pro.list"
```
Comment (eleo007) [inline docs/install-pro.md:31]: Nice-to-have: add how to get `OPERATING_SYSTEM` :  `lsb_release -sc`

File: docs/install-pro.md [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+            deb http://repo.percona.com/private/[CLIENTID]-[TOKEN]/psmysql-80-pro/apt/ OPERATING_SYSTEM main
```
Comment (eleo007) [inline docs/install-pro.md]: Change `psmysql-80-pro` to `ps-80-pro` : 
`deb http://repo.percona.com/private/[CLIENTID]-[TOKEN]/ps-80-pro/apt/ OPERATING_SYSTEM main`

File: docs/install-pro.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+            ```
+
+    === "On RHEL and derivatives"
+
+        Create the `/etc/yum.repos.d/psmysql-pro.repo` configuration file with the following contents
+
+        ```ini title="/etc/yum.repos.d/psmysql-pro.repo"
+        [psmysql-8.0-pro]
+        name=PSMYSQL_8.0_PRO
+        baseurl=http://repo.percona.com/private/[CLIENTID]-[TOKEN]/psmysql-80-pro/yum/main/$releasever/RPMS/x86_64
```
Comment (eleo007) [inline docs/install-pro.md]: Here also `ps-80-pro`

File: docs/psmysql-pro.md [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+
+[Become a Percona Customer](https://www.percona.com/about/contact){.md-button}
+
+## Capabilities
+
+Find the list of capabilities available in Percona Server for MySQL Pro:
+
+| Name                                | Version added | Description  | 
+| ----------------------------------- | ------------- | -------------
+| [FIPS support ](fips.md)| [8.0.35-28](release_notes/8.0.35-28.md) | Enables all commercial cloud service providers who want to sell and expand their footprint for US government entities. |
```
Comment (eleo007) [inline docs/psmysql-pro.md]: It is 8.0.35-27

File: docs/source-tarball.md [hunk 23]
```diff
@@ -1,5 +1,7 @@
 # Install Percona Server for MySQL from a source tarball
 
+<!----ask Eleonora whether we need to add here an instruction for PRO bulds---->
```
Comment (eleo007) [inline docs/source-tarball.md]: There will be no source tarballs for pro

File: docs/upgrade-pro.md [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+Request the access to the pro repository from Percona Support. You will receive the client ID and the access token which you use when downloading the packages.
+
+## Procedure
+
+1. Stop the `mysql` server
+
+    ```{.bash data-prompt="$"}
+    $ sudo systemctl stop mysql
+    ```
+2. Configure the repository
```
Comment (eleo007) [inline docs/upgrade-pro.md]: I would change steps (changes places of 1 and 2 ) + update with changes from install step

File: docs/upgrade-pro.md (hunk 14 above)
Comment (eleo007) [reply docs/upgrade-pro.md:56]: On deb nothing is needed. Just install pro after the regular

File: docs/upgrade-pro.md (hunk 15 above)
Comment (eleo007) [reply docs/upgrade-pro.md]: Here --allowerasing should be passed:
`sudo yum install --allowerasing percona-server-server-pro`

**Review by eleo007 — CHANGES_REQUESTED** [review]

File: docs/compile-percona-server.md [hunk 25]
```diff
@@ -1,6 +1,8 @@
 # Compile Percona Server for MySQL 8.0 from source
 
-The following instructions install Percona Server for MySQL 8.0.
+<!----ask Eleonora whether we need to add here an instruction for PRO bulds---->
```
Comment (eleo007) [inline docs/compile-percona-server.md]: AFAIK - no, @percona-ysorokin ?

Comment (percona-ysorokin) [reply docs/compile-percona-server.md]: Currently there is no difference in building Pro version (apart from specifying a different VERSION_COMMENT parameter)
So, I would also say "no".

Comment (alina-derkach-oaza) [reply docs/compile-percona-server.md]: Thank you. Removing the comment.

File: docs/upgrade-pro.md (hunk 24 above)
Comment (alina-derkach-oaza) [reply docs/upgrade-pro.md]: Updated

File: docs/source-tarball.md (hunk 23 above)
Comment (alina-derkach-oaza) [reply docs/source-tarball.md]: removed the comment

File: docs/psmysql-pro.md (hunk 22 above)
Comment (alina-derkach-oaza) [reply docs/psmysql-pro.md]: Updated

File: docs/install-pro.md (hunk 20 above)
Comment (alina-derkach-oaza) [reply docs/install-pro.md]: Updated

File: docs/install-pro.md (hunk 21 above)
Comment (alina-derkach-oaza) [reply docs/install-pro.md]: Updated

File: docs/install-pro.md (hunk 19 above)
Comment (alina-derkach-oaza) [reply docs/install-pro.md:31]: Updated

File: docs/install-pro.md (hunk 18 above)
Comment (alina-derkach-oaza) [reply docs/install-pro.md]: Updated

File: docs/fips.md (hunk 17 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: Updated

File: docs/fips.md (hunk 16 above)
Comment (alina-derkach-oaza) [reply docs/fips.md]: Updated

File: docs/upgrade-pro.md (hunk 15 above)
Comment (alina-derkach-oaza) [reply docs/upgrade-pro.md]: Updated

File: docs/binary-tarball-install.md [hunk 26]
```diff
@@ -4,10 +4,20 @@ A binary tarball contains a group of files, including the source code, bundled t
...
-Fetch and extract the correct binary tarball. For example for *Debian 10*:
+Fetch and extract the correct binary tarball. For example, for Debian 10:
 
 ```{.bash data-prompt="$"}
 $ wget https://downloads.percona.com/downloads/Percona-Server-8.0/Percona-Server-8.0.26-16/binary/tarball/Percona-Server-8.0.26-16-Linux.x86_64.glibc2.12.tar.gz
 ```
+
+## Install Percona Server for MySQL Pro from a binary tarball
+
+You can download the required binary tarball for Percona Server for MySQL Pro using your `CLIENTID` and `TOKEN` in the following link https://repo.percona.com/private/[CLIENTID]-[TOKEN]/ps-80-pro/.
```
Comment (eleo007) [inline docs/binary-tarball-install.md]: https://repo.percona.com/private/[CLIENTID]-[TOKEN]/ps-80-pro/tarballs/

File: docs/binary-tarball-install.md [hunk 27]
```diff
@@ -4,10 +4,20 @@ A binary tarball contains a group of files, including the source code, bundled t
...
+
+Fetch and extract the correct binary tarball using your `CLIENTID` and `TOKEN`. For example, for Debian 12:
```
Comment (eleo007) [inline docs/binary-tarball-install.md]: Let's better give an example for ol9 as we announce that Deb12 ad Jammy are more for testing

wget https://repo.percona.com/private/[CLIENTID]-[TOKEN]/ps-80-pro/tarballs/Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.34.tar.gz

File: docs/fips.md [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+
+## Prerequisites
+
+To prepare Percona Server for MySQL for FIPS certification, do the following:
+
+* Check that your operating system includes FIPS pre-approved OpenSSL library in version 3.0.x or higher. The following distributions includes FIPS pre-approved OpenSSL library in version 3.0.x or higher:
+
+    * RedHat Enterprise Linux 9
+
+    * Oracle Linux 9
```
Comment (eleo007) [inline docs/fips.md:15]: >    * RedHat Enterprise Linux 9

>    * Oracle Linux 9

@percona-ysorokin maybe it is better to write RedHat Enterprise Linux 9 and derivatives here?

File: docs/install-pro.md [hunk 29]
```diff
@@ -0,0 +1,68 @@
+# Install Pro packages of Percona Server for MySQL
```
Comment (eleo007) [inline docs/install-pro.md]: Please add the list of packages that are available for PRO:
packages deb:
      - percona-server-server-pro
      - percona-server-test-pro
      - percona-server-client-pro
      - percona-server-pro-dbg
      - percona-server-pro-source
      - percona-server-rocksdb-pro
       -percona-mysql-router-pro
       -[percona-server-pro-common]
packages rpm:
    percona-server-server-pro
    percona-server-shared-pro
    percona-server-client-pro
    percona-server-pro-debuginfo
    percona-server-rocksdb-pro
    percona-server-test-pro
    percona-server-devel-pro
    percona-mysql-router-pro

File: docs/upgrade-pro.md [hunk 30]
```diff
@@ -0,0 +1,65 @@
+# Upgrade to Percona Server for MySQL Pro
```
Comment (eleo007) [inline docs/upgrade-pro.md:1]: Please add the list of packages that are available for PRO:
packages deb:
- percona-server-server-pro
- percona-server-test-pro
- percona-server-client-pro
- percona-server-pro-dbg
- percona-server-pro-source
- percona-server-rocksdb-pro
-percona-mysql-router-pro
-[percona-server-pro-common]
packages rpm:
percona-server-server-pro
percona-server-shared-pro
percona-server-client-pro
percona-server-pro-debuginfo
percona-server-rocksdb-pro
percona-server-test-pro
percona-server-devel-pro
percona-mysql-router-pro

**Review by eleo007 — CHANGES_REQUESTED** [review]

File: docs/fips.md (hunk 28 above)
Comment (percona-ysorokin) [reply docs/fips.md:15]: @eleo007 In general I agree, we should. But have we ever tested this on real RHEL9?

Comment (eleo007) [reply docs/fips.md:15]: I have tested on OL9

**Comment by alina-derkach-oaza** [comment]
@eleo007 Please clarify what files contain the following packages:
percona-server-rocksdb-pro
percona-mysql-router-pro
percona-server-shared-pro
percona-server-devel-pro

**Comment by eleo007** [comment]
> @eleo007 Please clarify what files contain the following packages: percona-server-rocksdb-pro percona-mysql-router-pro percona-server-shared-pro percona-server-devel-pro

shared and devel are the same as for regular server rpms https://docs.percona.com/percona-server/8.0/yum-files.html?h=percona+server :
percona-server-shared-pro	Client shared library.
percona-server-devel	Header files needed to compile software using the client library.

As for:
percona-server-rocksdb-pro - installs rocksdb 
percona-mysql-router-pro - this is mysql router (actually this is the separate tool and not fully part of the percona-server, not sure it should be in the files list)

By the way they are also available for non-pro, I am not sure why we do not have them in regular docs

File: docs/binary-tarball-names.md [hunk 31]
```diff
@@ -43,6 +43,24 @@ If the `glibc2` version from your operating system is not listed, then this Perc
...
+    | Ubuntu 22.04         | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.35.tar.gz         | glibc2.35      |
+    | Ubuntu 22.04         | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.35-minimal.tar.gz | glibc2.35      |
+    | Red Hat Enterprise 9 | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.34.tar.gz         | glibc2.34      |
+    | Red Hat Enterprise 9 | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.34-minimal.tar.gz | glibc2.34      |
+
+    The types of files are the following:
+
+    | Type | Name | Description |
+    |---|---|---|
+    | Full | Percona-Server-<version-number>-Linux.x86_64.<glibc2-version>.tar.gz | Contains all files available |
```
Comment (eleo007) [inline docs/binary-tarball-names.md]: Maybe Here Percona-Server-Pro-<version-number>?

File: docs/fips.md [hunk 32]
```diff
@@ -0,0 +1,78 @@
...
+
+    * `OFF` - disables FIPS mode
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
+     
+    The MySQL server operation in FIPS mode does not depend on which crypto module (regular or FIPS-approved) is set as the default in the OpenSSL configuration file. The server always respects the value of `--ssl-fips-mode` server command line option (`OFF`, `ON`, or `STRICT`). The `ssl_fips_mode` global system variable is read-only and cannot be changed at runtime.
+  
+### Enable the FIPS mode
+
+To enable the FIPS mode, pass `--ssl-fips-mode=ON` to mysqld as a command line argument or add `ssl-fips-mode=ON` to the configuration file. Ignore the warning that the `--ssl-fips-mode` client/server option is deprecated.
```
Comment (eleo007) [inline docs/fips.md]: --ssl-fips-mode can also be STRICT. `--ssl-fips-mode=ON` or `--ssl-fips-mode=STRICT`

File: docs/install-pro.md [hunk 33]
```diff
@@ -0,0 +1,73 @@
...
+
+            ```{.bash .data-prompt="$"}
+            $ sudo apt update
+            ```
+        
+        5. Install Percona Server for MySQL packages
+        
+            ```{.bash .data-prompt="$"}
+            $ sudo apt install -y percona-server-server-pro
+            ```
```
Comment (eleo007) [inline docs/install-pro.md:45]: Please also add here link to other packages for PRO

File: docs/install-pro.md [hunk 34]
```diff
@@ -0,0 +1,73 @@
...
+            name=PS_8.0_PRO
+            baseurl=http://repo.percona.com/private/[CLIENTID]-[TOKEN]/ps-80-pro/yum/main/$releasever/RPMS/x86_64
+            enabled=1
+            gpgkey = https://repo.percona.com/yum/PERCONA-PACKAGING-KEY
+            ```
+
+        2. Install Percona Server for MySQL packages
+        
+            ```{.bash .data-prompt="$"}
+            $ sudo yum install -y percona-server-server-pro
```
Comment (eleo007) [inline docs/install-pro.md:64]: Add link to all packages

File: docs/upgrade-pro.md [hunk 35]
```diff
@@ -0,0 +1,84 @@
...
+    ```{.bash data-prompt="$"}
+    $ sudo systemctl stop mysql
+    ```
+
+3. Install Percona Server for MySQL Pro packages
+
+    === "On Debian and Ubuntu"
+
+        ```{.bash .data-prompt="$"}
+        $ sudo apt install -y percona-server-server-pro
```
Comment (eleo007) [inline docs/upgrade-pro.md:56]: Please also add the link to other packages

**Review by eleo007 — CHANGES_REQUESTED** [review]

File: docs/fips.md [hunk 36]
```diff
@@ -0,0 +1,78 @@
...
+
+Percona Server for MySQL uses the same variables and values as MySQL. Percona Server for MySQL enables control of FIPS mode on the server side and the client side:
+
+* The `ssl_fips_mode` system variable shows whether the server operates in FIPS mode. This variable is disabled by default.
+
+    The `ssl_fips_mode` system variable has these values:
+
+    * `0` - disables FIPS mode
+    * `1` - enables FIPS mode
+    * `2` - enables `strict` FIPS mode. This value provides more restrictions than the  `1 ` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `1` or `2` depends on the OpenSSL version.
```
Comment (patrickbirch) [inline docs/fips.md]: The exact behavior of the `strict` FIPS mode depends on the OpenSSL version. The server only specifies the FIPS value to OpenSSL.

File: docs/fips.md [hunk 37]
```diff
@@ -0,0 +1,78 @@
...
+
+* The `--ssl-fips-mode` client/server option controls whether a given MySQL client operates in FIPS mode. This setting does not change the server setting. This option is disabled by default.
+
+    The `--ssl-fips-mode` client/server option has these values:
+
+    * `OFF` - disables FIPS mode
+    * `ON` - enables FIPS mode
+    * `STRICT` - enables `strict` FIPS mode. This value provides more restrictions than the `ON` value. However, MySQL itself does not have any FIPS-specific code other than to specify the FIPS mode value to OpenSSL. The exact behavior of FIPS mode for `ON` or `STRICT` depends on the OpenSSL version.
```
Comment (patrickbirch) [inline docs/fips.md]: Same as above.
We don't know if any restrictions are applied to the client because that's dependent on the OpenSSL version.

**Review by eleo007 — APPROVED** [review]

File: docs/binary-tarball-names.md [hunk 38]
```diff
@@ -38,10 +38,28 @@ If the `glibc2` version from your operating system is not listed, then this Perc
...
+=== "Pro builds"
+
+    The following binary tarballs are available for Percona Server for MySQL Pro builds from version 8.0.35-27 and later.
+
+    | Platform             | Percona Server for MySQL Pro tarball name                          | glibc2 version |
+    |----------------------|--------------------------------------------------------------------|----------------|
+    | Ubuntu 22.04         | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.35.tar.gz         | glibc2.35      |
+    | Ubuntu 22.04         | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.35-minimal.tar.gz | glibc2.35      |
+    | Red Hat Enterprise 9 | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.34.tar.gz         | glibc2.34      |
+    | Red Hat Enterprise 9 | Percona-server-Pro-8.0.35-27-Linux.x86_64.glibc2.34-minimal.tar.gz | glibc2.34      |
```
Comment (adivinho) [inline docs/binary-tarball-names.md]: The "server" word starts from a capital.
Screenshot 2024-02-26 at 14 52 25

**Review by adivinho — CHANGES_REQUESTED** [review]

**Review by adivinho — APPROVED** [review]

**Review by patrickbirch — APPROVED** [review]
Good

**Pull request merged**